sdk/Sdk/FunctionMetadataGenerator.cs

```diff
@@ -462,5 +487,55 @@ private static bool TryGetBaseAttributeType(CustomAttribute attribute, string ba
 
             return false;
         }
+
+        private static bool IsIterableCollection(TypeReference type, out DataType dataType)
+        {
+            // Array and not byte array 
+            bool isArray = type.IsArray && !string.Equals(type.FullName, typeof(byte[]).FullName, StringComparison.Ordinal);
+            if (isArray)
+            {
+                TypeSpecification? typeSpecification = type as TypeSpecification;
+                if (typeSpecification is not null)
+                {
+                    dataType = GetDataTypeFromType(typeSpecification.ElementType.FullName);
+                    return true;
+                }
+            }
+
+            dataType = DataType.Undefined;
+            return false;
+        }
+
+        private static DataType GetDataTypeFromType(string fullName)
+        {
+            if (IsStringType(fullName))
+            {
+                return DataType.String;
+            }
+            else if (IsBinaryType(fullName))
+            {
+                return DataType.Binary;
+            }
+
+            return DataType.Undefined;
+        }
+
+        private static bool IsStringType(string fullName)
+        {
+            return string.Equals(fullName, typeof(string).FullName, StringComparison.Ordinal);
```

Comment:
nit: but as an optimization, the typeof(<type>) name can be resolved and reused.